src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -61,6 +61,50 @@ ucs_status_t uct_rdmacm_cm_reject(struct rdma_cm_id *id)
     return UCS_OK;
 }
 
+ucs_status_t uct_rdmacm_cm_get_cq(uct_rdmacm_cm_t *cm, struct ibv_context *verbs,
+                                  const char *device_name, struct ibv_cq **cq)
+{
+    khiter_t iter = kh_get(uct_rdmacm_cm_cqs, &cm->cqs, device_name);
+    int ret;
+
+    if (iter != kh_end(&cm->cqs)) {
+        *cq = kh_val(&cm->cqs, iter);
+        return UCS_OK;
+    }
+
+    /* Create a dummy completion queue */
+    *cq = ibv_create_cq(verbs, 1, NULL, NULL, 0);
+    if (cq == NULL) {
+        ucs_error("ibv_create_cq() failed: %m");
+        return UCS_ERR_IO_ERROR;
+    }
+
+    iter = kh_put(uct_rdmacm_cm_cqs, &cm->cqs, device_name, &ret);
+    if (iter == kh_end(&cm->cqs)) {
+        ibv_destroy_cq(*cq);
```

Comment:
pls check return value

Reply:
I don't think it makes sense here on out-of-memory flow, usually this is an unrecoverable error.